jina/flow/base.py

```diff
@@ -912,7 +914,7 @@ def use_grpc_gateway(self, port: Optional[int] = None):
 
     def _switch_gateway(self, gateway: str, port: int):
         restful = gateway == 'RESTRuntime'
-        client = WebSocketClient if gateway == 'RESTRuntime' else Client
+        client = WebSocketClient if gateway == 'RESTRuntime' else GrpcClient
```

Comment:
Why can we not use `Client` here passing restful?

Reply:
@deepankarm we can, while it does not make a lot difference since `Client` is actually initialising `GrpcClient` and `WebSocketClient`

Reply:
then lets have a single poijt of entry, change it to Client with the proper argument, u can use partial functions if needed

Reply:
done

Reply:
I meant, not using `GrpcClient` & `WebSocketClient` at all in `flow/base.py` and using `Client` directly. Not important though.

Reply:
gave a try, and I broke a lot of tests. I'll look into if I can optimise this part in a future PR.

Reply:
> and I broke a lot of tests

I believe the reason was because your `Client()` function does not accept the full arguments that CLI defined. This oversimplification causes incomplete setup of the client and yields the broken test.

https://github.com/jina-ai/jina/pull/2565#discussion_r647896285